### PR TITLE
Support for gcc 13

### PR DESCRIPTION
**Pull request by BrunoASMauricio**
Fix for https://github.com/foss-for-synopsys-dwc-arc-processors/buildroot/issues/37

Following the commented suggestion, cherry-picked commits from https://github.com/RomainNaour/buildroot/commits/toolchain-builder-2023.05

GCC 13.0 can now be selected and the builds follow through with success

### CODE REVIEW

File: package/gcc/Config.in.host [hunk 1]
```diff
@@ -58,6 +58,19 @@ config BR2_GCC_VERSION_12_X
...
+	bool "gcc 13.x"
+	# powerpc spe support has been deprecated since gcc 8.x.
+	# https://gcc.gnu.org/ml/gcc/2018-04/msg00102.html
+	depends on !BR2_powerpc_SPE
+	# uClibc-ng broken on sparc due to recent gcc changes
+	# that need to be reverted since gcc 8.4, 9.3 and 10.1.
+	# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98784
+	depends on !BR2_sparc
+	# ARC HS48 rel 31 only supported by gcc arc fork.
+	depends on !BR2_archs4x_rel31
```
Comment (abrodkin) [inline package/gcc/Config.in.host]: ARC HS4x release 3.1 is supported in the upstream GCC 13, see https://gcc.gnu.org/git/?p=gcc.git;a=commit;h=7501eec65c60701f72621d04eeb5342bad2fe4fb.

**Review by abrodkin — APPROVED** [review]
Thanks for the fix. See my tiny note on the HS4x release 3.1, but it's more of a rant to or note to Romain, so we may have his patches as they are.

**Comment by pavelvkozlov** [comment]
Looks good for me.  @abrodkin I think we should rebase our arc64 branch to the latest upstream. Now in upstream buildroot uses glibc 2.37 without any patches and its good for us (no need to delete patches) and I think it is better to rebase before merge of this PR.

**Review by pavelvkozlov — APPROVED** [review]

**Comment by BrunoASMauricio** [comment]
No longer necessary with the latest changes

**Pull request closed without merging**